jangjuseong/app.js

```diff
@@ -50,14 +50,148 @@ app.post('/users/signup', async (req, res) => {
         ?, 
         ?, 
         ?
-        );
+      );
     `,
     [name, email, password, profileImage]
   );
 
   res.status(201).json({ message: 'userCreated' });
 });
 
+app.post('/posts', async (req, res) => {
+  const { title, content, postImgUrl, userId } = req.body;
+
+  await mysqlDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      post_image_url,
+      user_id
+      )
+      VALUES (
+        ?,
+        ?,
+        ?,
+        ?
+      );
+        `,
+    [title, content, postImgUrl, userId]
+  );
+
+  res.status(201).json({ message: 'postCreated' });
+});
+
+app.get('/posts', async (req, res) => {
+  const posts = await mysqlDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id;
+    `
+  );
+
+  res.status(200).json({ posts });
+});
+
+app.get('/posts/user/:userId', async (req, res) => {
+  const { userId } = req.params;
+
+  await mysqlDataSource
+    .query(
+      `SELECT
+    u.id AS userId,
+    u.profile_image AS userProfileImage,
+    pi.post_informations AS postings
+    FROM
+      users u
+    INNER JOIN (
+      SELECT
+        user_id,
+        JSON_ARRAYAGG(
+          JSON_OBJECT (
+            "postingId", id,
+            "postingImageUrl", post_image_url,
+            "postingContent", content
+          )
+        ) AS post_informations
+      FROM
+        posts
+      GROUP BY
+        user_id
+    ) pi 
+      ON pi.user_id = u.id
+    WHERE
+      u.id = ${userId};
+    `
+    )
+    .then((data) => res.status(200).json({ data }));
```

Comment:
- async/await 키워드를 사용하고 있기 때문에 then() 함수를 통해 비동기 함수를 처리할 필요가 없습니다.
- SQL 쿼리 포매팅을 다음과 같이 수정해주면 구조 파악이 용이해지고 가독성도 좋아질 것 같습니다.

```suggestion
  const [result] = await mysqlDataSource.query(`
    SELECT
      u.id AS userId,
      u.profile_image AS userProfileImage,
      pi.post_informations AS postings
    FROM users u
    INNER JOIN (
      SELECT
        user_id,
        JSON_ARRAYAGG(
          JSON_OBJECT (
            "postingId", id,
            "postingImageUrl", post_image_url,
            "postingContent", content
          )
        ) AS postInformations
      FROM posts
      GROUP BY user_id
    ) pi 
      ON pi.user_id = u.id
    WHERE u.id = ?;
    `, [ userId ])
    return res.status(200).json({ data: result }));
```